protocol/lib/metrics/util.go

```diff
@@ -80,3 +80,18 @@ func GetCallbackMetricFromCtx(ctx sdk.Context) string {
 
 	return DeliverTx
 }
+
+func ContextuallySetGaugeWithLabels(
```

Comment:
There are other telemetry function that we use often e.g. `IncrCounterWithLabels`

Could we find a more generic pattern?
```
type TelemetryFunc func(keys []string, val float32, labels []gometrics.Label)

func ContextuallyEmitTelemetry(
	ctx types.Context,
	telemetryFunc TelemetryFunc,
	allowedModes []types.ExecMode,
 ......
) {
	contextExecMode := ctx.ExecMode()
	for _, mode := range allowedModes {
		if contextExecMode == mode {
			telemetryFunc(keys, val, labels)
			break // Exit once the correct mode is found and function is executed
		}
	}
}
```

Reply:
I see two common method signatures. Added one for each of them